packages/api/es/collections.js

```diff
@@ -76,12 +80,81 @@ class Collection extends BaseSearch {
     });
   }
 
-  async query() {
-    const res = await super.query();
+  /**
+   * Get a list of collection ids that have granules. If time params
+   * are specified the query will return collections that have granules that have been updated
+   * in that time frame.
+   *
+   * @returns {Promise<Array<string>>} - list of collection ids with active granules
+   */
+  async aggregateActiveGranuleCollections() {
+    if (!this.client) {
+      this.client = await this.constructor.es();
+    }
+
+    // granules
+    const searchParams = this._buildSearch();
+    searchParams.size = 0;
+    delete searchParams.from;
+    searchParams.type = 'granule';
+
+    searchParams.body.aggs = {
+      collections: {
+        terms: {
+          field: 'collectionId'
+        }
+      }
+    };
+
+    const searchResults = await this.client.search(searchParams)
+      .then((response) => response.body);
+
+    return searchResults.aggregations.collections.buckets.map((b) => b.key);
+  }
+
+  /**
+   * Add collection granule stats to collection query results
+   *
+   * @param {Array<Object>} collectionResults - collection query results
+   * @returns {Promise<Array<Object>>} - collectionResults updated with granule stats
+   */
+  addStatsToCollectionResults(collectionResults) {
+    const ids = collectionResults.filter((r) => r.name)
+      .map((c) => constructCollectionId(c.name, c.version));
+    return this.getStats(collectionResults, ids);
+  }
+
+  /**
+   * Perform a collection query to return collections that have granules. If time params
+   * are specified the query will return collections that have granules that have been updated
+   * in that time frame.
+   *
+   * @returns {Promise<Object>} - query result object containing collections and their granule stats
+   */
+  async queryCollectionsWithActiveGranules() {
+    const collectionIds = await this.aggregateActiveGranuleCollections();
+
+    const searchParams = this._buildSearch();
+    searchParams.body.query = {
+      constant_score: {
+        filter: {
+          terms: {
+            _id: collectionIds
+          }
+        }
+      }
+    };
+
+    const res = await this.query(searchParams);
+
+    return res;
+  }
+
+  async query(searchParamsOverride = null) {
```

Comment:
I think this default assignment to `null` is unnecessary. It would default to `undefined` if not specified and the logic in `Basesearch.query` would behave the same